spec/requests/products_controller_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe ProductsController, type: :request do
+
+  describe 'POST products#add_to_cart' do
+    before(:all) do
+      product = create(:product)
+
+      post add_to_cart_product_path(product.id)
+    end
+
+    context 'unauthorized user' do
```

Comment:
you should also add a context for a successful test, not just for unauthorized user. also, in that case you should structure:

```ruby
describe 'POST whatever' do
  context 'unauthorized user'
    before do
      # create the product, do any operations you need
    end

    it 'asks user for login' do
      expect..
    end
  end

  context 'successful add to cart'
    before do
      # create the product, do any operations you need
    end

    it 'successfully adds product to cart' do
      expect..
    end
  end
end
```